### PR TITLE
Improve source marker placement for line_length_linter

### LINKED ISSUE #735: Move line_length_linter column to one past `length`
Oh sorry, I misunderstood. IIUC the `^~~~` tag is only built from ranges. So `column_number` should be fine to update. But perhaps best to keep as a follow-up to keep the PR more focused.

_Originally posted by @MichaelChirico in https://github.com/jimhester/lintr/issues/734#issuecomment-772792089_

Makes the source marker more useful for fixing the lint, since it places the cursor at the spot where a line break should occur instead of at the start of the line.

### PR DESCRIPTION
**Pull request by AshesITR**
fixes #735

 - [x] implement
 - [x] add tests for column_number
 - [x] add NEWS bullet

I also cleaned up the test code by using strrep() while I was working on it.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -43,6 +43,8 @@
   (#660, @AshesITR)
 * Fixed `spaces_left_parentheses_linter` sporadically causing warnings (#654, #674, @AshesITR)
 * Fixed `line_length_linter` causing duplicate lints for lines containing multiple expressions (#681, #682, @AshesITR)
+* `line_length_linter` now places the source marker at the end of the affected line to improve user experience during
```
Comment (MichaelChirico) [inline NEWS.md:46]: reworded, LMK WYT

File: tests/testthat/test-line_length_linter.R [hunk 2]
```diff
@@ -1,40 +1,48 @@
...
-  expect_lint("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
-    NULL,
-    line_length_linter(80))
-
-  expect_lint("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
-    "Lines should not be more than 80 characters",
-    line_length_linter(80))
-
+  expect_lint("blah", NULL, linter)
+  expect_lint(strrep("x", 80L), NULL, linter)
```
Comment (MichaelChirico) [inline tests/testthat/test-line_length_linter.R:6]: good usage of `strrep`!

**Review by MichaelChirico — DISMISSED** [review]
LGTM, feel free to merge if you're OK w the NEWS item

**Comment by AshesITR** [comment]
I like the new wording in general, but it's not the end of the affected line, it's at the margin, maybe?
WDYT?

**Review by MichaelChirico — APPROVED** [review]

**Pull request merged**